geode-core/src/main/java/org/apache/geode/management/internal/operation/OperationHistoryManager.java

```diff
@@ -38,146 +33,88 @@
  */
 @Experimental
 public class OperationHistoryManager {
-  private final ConcurrentMap<String, OperationInstance> history;
   private final long keepCompletedMillis;
+  private final OperationStateDistributionService operationStateDistributionService;
 
   /**
    * set a default retention policy to keep results for 2 hours after completion
    */
-  public OperationHistoryManager() {
-    this(2, TimeUnit.HOURS);
+  public OperationHistoryManager(
+      OperationStateDistributionService operationStateDistributionService) {
+    this(2, TimeUnit.HOURS, operationStateDistributionService);
   }
 
   /**
    * set a custom retention policy to keep results for X amount of time after completion
    */
-  public OperationHistoryManager(long keepCompleted, TimeUnit timeUnit) {
```

Comment:
Consider using Java's very nice `Duration` and `Instant` classes to carry information about durations and instants.

Reply:
constructor now takes a Duration instead of a long and a TimeUnit